Source/WebCore/editing/Editor.cpp

```diff
@@ -1233,8 +1248,15 @@ void Editor::appliedEditing(CompositeEditCommand& command)
     ASSERT(command.composition());
     Ref composition = *command.composition();
     VisibleSelection newSelection(command.endingSelection());
+    auto startRoot = composition->startingRootEditableElement();
+    auto endRoot = composition->endingRootEditableElement();
+    auto inputType = command.inputEventTypeName();
+    auto data = command.inputEventData();
 
-    notifyTextFromControls(composition->startingRootEditableElement(), composition->endingRootEditableElement());
+    if (inputType == "insertText"_s || inputType == "insertParagraph"_s || inputType == "insertLineBreak"_s)
```

Comment:
Why are we checking this? That's not mentioned in the commit message?

Reply:
>>     if (inputType == "insertText"_s || inputType == "insertParagraph"_s || inputType == "insertLineBreak"_s)`
> Why are we checking this? That's not mentioned in the commit message?

That’s per step 4.6 of the algorithm at https://w3c.github.io/uievents/event-algo.html#fire%20key%20input%20events:
> If _inputType_ is not one of "insertText", "insertParagraph" or "insertLineBreak", then set _shouldFireTextInput_ to false.

I’ve updated the commit message to explicitly mention that.